ai/worker/docker.go

```diff
@@ -498,6 +499,13 @@ func (m *DockerManager) watchContainer(rc *RunnerContainer, borrowCtx context.Co
 		if failures >= maxHealthCheckFailures && time.Since(startTime) > pipelineStartGracePeriod {
 			slog.Error("Container health check failed too many times", slog.String("container", rc.Name))
 			m.destroyContainer(rc, false)
+			if rc.KeepWarm {
```

Comment:
I think this will not work, when the container is idle. For example, you can test locally:
1. Start Orchestrator with Warm runner => Orchestrator starts the container ALL GOOD
2. Wait 60s => `Container is idle, returning to pool container` ALL GOOD
3. Stop Container (simulate that it crashes) => It won't be recreated NOT GOOD

I think that maybe we should move this `warchContainer` (or create similar function), so that the warm container is ALWAYS healthchecked (and recreated) not only when it's not idle.

Reply:
Good catch, yeah, we only watch containers as they are being used for now. It is unlikely they will crash when idle, but I guess it's not too big of a refactor to change this behavior so I'll include that here.

Reply:
yeah, if in rush, then I'm ok merging this PR as it is. Otherwise, it would be good to restart also IDLE container. But right, it's unlikely they crash while not doing anything 🙃

Reply:
Ok, Merging this PR, and we can address this ☝️ as a separate PR.

Reply:
Good point. I think that yes, I'm just thinking about some cases like when we update the runner image, but hmm, then we can just manually stop the runner container (which we already do in ansible).

@victorges wdyt?

Reply:
That's a great idea!